src/Classes/PassiveTree.lua

```diff
@@ -417,6 +418,28 @@ local PassiveTreeClass = newClass("PassiveTree", function(self, treeVersion)
 		end
 	end
 
+	for name, keystone in pairs(self.keystoneMap) do
+		keystone.nodesInRadius = { }
+		for radiusIndex, radiusInfo in ipairs(data.jewelRadius) do
+			keystone.nodesInRadius[radiusIndex] = { }
+			local outerRadiusSquared = radiusInfo.outer * radiusInfo.outer
+			local innerRadiusSquared = radiusInfo.inner * radiusInfo.inner
+			for _, node in pairs(self.nodes) do
+				if (keystone.x and keystone.y) then
```

Comment:
I would put this `if` above the `for` just above it since `keystone` is know. It's more efficient CPU wise.